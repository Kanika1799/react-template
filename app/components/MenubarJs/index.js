/**
 *
 * MenubarJs
 *
 */

import React, { memo } from 'react';
// import PropTypes from 'prop-types'
import logo from '@images/logo.svg';
import invoice from '@images/ic-my-invoices.svg';
import dash from '@images/ic-dash.svg';
import styled from 'styled-components';
// import { FormattedMessage as T } from 'react-intl';
import 'antd/dist/antd.css';
import { Menu } from 'antd';
const MenuDiv = styled.div`
  width: 70px;
`;
function Menubar() {
  return (
    <MenuDiv data-testid="menubar-js">
      <Menu style={{ backgroundColor: '#e8f0fd', height: 600 }}>
        <Menu.Item style={{ height: 70, marginLeft: 0, marginTop: 20, padding: 5 }} key="1">
          <img src={logo} />
        </Menu.Item>
        <Menu.Item key="2" style={{ height: 70, marginTop: 80, padding: 10 }}>
          <img src={invoice} />
        </Menu.Item>
        <Menu.Item key="3" style={{ height: 70, marginTop: 60, padding: 10 }}>
          <img width="50" src={dash} />
        </Menu.Item>
      </Menu>
    </MenuDiv>
  );
}

Menubar.propTypes = {};

export default memo(Menubar);
