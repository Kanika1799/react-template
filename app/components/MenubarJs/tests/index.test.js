/**
 *
 * Tests for MenubarJs
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import MenubarJs from '../index';

describe('<MenubarJs />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<MenubarJs />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 MenubarJs component', () => {
    const { getAllByTestId } = renderWithIntl(<MenubarJs />);
    expect(getAllByTestId('menubar-js').length).toBe(1);
  });
});
