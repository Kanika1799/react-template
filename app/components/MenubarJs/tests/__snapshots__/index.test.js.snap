// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenubarJs /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="MenubarJs__MenuDiv-sc-9eruv1-0 lautxb"
      data-testid="menubar-js"
    >
      <ul
        class="ant-menu ant-menu-light ant-menu-root ant-menu-vertical"
        role="menu"
        style="background-color: rgb(232, 240, 253); height: 600px;"
      >
        <li
          class="ant-menu-item ant-menu-item-only-child"
          role="menuitem"
          style="height: 70px; margin-left: 0px; margin-top: 20px; padding: 5px;"
        >
          <img
            src="IMAGE_MOCK"
          />
        </li>
        <li
          class="ant-menu-item ant-menu-item-only-child"
          role="menuitem"
          style="height: 70px; margin-top: 80px; padding: 10px;"
        >
          <img
            src="IMAGE_MOCK"
          />
        </li>
        <li
          class="ant-menu-item ant-menu-item-only-child"
          role="menuitem"
          style="height: 70px; margin-top: 60px; padding: 10px;"
        >
          <img
            src="IMAGE_MOCK"
            width="50"
          />
        </li>
      </ul>
    </div>
  </div>
</body>
`;
